pkg/config/log.go

```diff
@@ -350,10 +350,28 @@ func parseShortFilePath(params string) seelog.FormatterFunc {
 }
 
 func extractShortPathFromFullPath(fullPath string) string {
-	// We want to trim the part containing the path of the project
-	// ie DataDog/datadog-agent/ or DataDog/datadog-process-agent/
-	slices := strings.Split(fullPath, "-agent/")
-	return slices[len(slices)-1]
+	shortPath := ""
+	if strings.Contains(fullPath, "-agent/") {
+		// We want to trim the part containing the path of the project
+		// ie DataDog/datadog-agent/ or DataDog/datadog-process-agent/
+		slices := strings.Split(fullPath, "-agent/")
+		shortPath = slices[len(slices)-1]
+	} else {
+		// For logging from dependencies, we want to log e.g.
+		// "go.opentelemetry.io/collector@v0.35.0/service/collector.go"
+		slices := strings.Split(fullPath, "/")
+		pkgNameIndex := len(slices) - 1
+		for ; pkgNameIndex > 0; pkgNameIndex-- {
+			if strings.Contains(slices[pkgNameIndex], "@") {
+				if pkgNameIndex > 0 {
+					pkgNameIndex--
```

Comment:
Can't a dependency be in a subdirectory, eg `go.opentelemetry.io/my_awesome_subdirectory/collector@v0.35.0`

Reply:
You're right, I'll have to figure out a smarter way to do this (if there is one).

Reply:
Ok, so I've found out that this could actually also be just e.g. `go.opencensus.io@v0.23.0`, which will also break this. So I think the safest way is going to be to just preserve throw away everything before the part with "@". The "@" will let us know that this file is coming from dependency and hopefully will be unique enough to make debugging easy.